enterprise/coderd/coderd.go

```diff
@@ -252,6 +252,17 @@ func (api *API) updateEntitlements(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	if entitlements.RequireTelemetry && !api.DeploymentConfig.Telemetry.Enable.Value {
+		// We can't fail because then the user couldn't remove the offending
+		// license w/o a restart.
+		api.entitlements.Errors = []string{
+			"License requires telemetry but telemetry is disabled",
+		}
+		api.Logger.Error(ctx, "license requires telemetry enabled")
+		return nil
+	}
+
```

Comment:
If you return early it doesn't seem like any of the enterprise features would work, if that's your goal. I think it'd probably be better to just append to the entitlements created above instead of overwriting and let the rest of the function run.